src/query/parser/promql/parse.go

```diff
@@ -285,6 +322,11 @@ func (p *parseState) walk(node pql.Node) error {
 		for i := 0; i < numExpectedValues; i++ {
 			argType := argTypes[i]
 			expr := expressions[i]
+			s, ok := expr.(*pql.SubqueryExpr)
+			if ok {
```

Comment:
Seems like this check is done before a bunch of others.

Might it read cleaner just combining into a switch with the other type switches?

e.g.
```go
switch e := expr.(type) {
case *pql.SubqueryExpr:
  argValues = append(argValues, s.Range)
case *pql.MatrixSelector:
  argValues = append(argValues, e.Range)
}

switch argType {
case pql.ValueTypeScalar:
  val, err := resolveScalarArgument(expr)
  if err != nil {
  	return err
  }
  argValues = append(argValues, val)
case pql.ValueTypeString:
  stringValues = append(stringValues, expr.(*pql.StringLiteral).Val)
default:
  if err := p.walk(expr, opts); err != nil {
    return err
  }
}
}
```